go-controller/pkg/kube/kube.go

```diff
@@ -23,6 +25,9 @@ type Interface interface {
 	SetAnnotationsOnPod(namespace, podName string, annotations map[string]string) error
 	SetAnnotationsOnNode(node *kapi.Node, annotations map[string]interface{}) error
 	SetAnnotationsOnNamespace(namespace *kapi.Namespace, annotations map[string]string) error
+	SetTaintOnNode(nodeName string, taint *kapi.Taint) error
```

Comment:
Note: These bits (set, remove patch taints) are also needed for the work being done in https://github.com/ovn-org/ovn-kubernetes/pull/2326. I have tried to align it as much as possible to what is required there so that both can leverage it, but a few things might need to be changed for the other PR or vice versa based on which gets in first.